### ISSUE TITLE
Support ad loading performance report

### ISSUE THREAD

**Issue by zhouyx** [ampproject/amphtml#5200]
We have the `render-start` API for ad now, it would give us good information on when ad is loaded. We could work on collection ad loading performance to get a better idea on how long it takes for an ad to load, then calibrate our timeout value. 
#4746 is a staring PR for this. However we defer it because I need a better solution to separate ad loading data from page loading data, and group them by ad types.

Current plan is to create new ad loading data category in parallel to the existing page loading data category. We may also need to create multiple new categories dynamically by ad type (we will start with `doubleclick` and `adsense` now). which means the creation should only take place once runtime layout an ad of certain type. Ideally we want to call `performance.tickDelta(category, label, value)`

reporting structure would be something like: 

```
page loading: {label1: data, label1: data, label1: data, label2: data .......}
doubleclick loading: {rs: data, rs:data, rs:data, otherlabel: data .......}
adsense loading: {rs: data, rs: data, ......}
```

Will work on this issue when we have support on the reporting API.

This is part of the task for #4586.

@erwinmombay  @lannka

**Comment by jpettitt** [comment]
Will this data be available via `amp-analytics` to give publishers an overview of all the ads on a page?

**Comment by jasti** [comment]
@jpettitt That's a good idea that we've been mulling ourselves. CC @avimehta and @rudygalfi for thoughts.

**Comment by avimehta** [comment]
This sounds like a great idea. @zhouyx lets sync up on what can be done here to asupport the variables in amp-analytics.

**Comment by lannka** [comment]
Exposing this data further to Analytics can be a separate topic. It will not be as trivial as adding a new metric, because all the current metrics are page level, and the new metric will be ad level. I can imagine we probably will need to introduce new triggering method like 'ad_load', and mechanism to do the URL replacement without having multiple ads load racing each other.

Let's keep this task just for Viewer only. We can discuss the "for Analytics" requirement in a separate thread.

**Comment by adelinamart** [comment]
Assigning this to Vamsee to decide if this something that will be needed to be added on the roadmap or close.

**Comment by jasti** [comment]
@zhouyx  @lannka Does it make sense to have a holistic way of tracking and reporting ad performance data? This will tie into the work for reducing 3P ad latency too. Thoughts?

**Comment by lannka** [comment]
will be addressed in #7500
